src/psyclone/psyir/nodes/call.py

```diff
@@ -72,38 +72,65 @@ def __init__(self, routine, parent=None):
+        its arguments.
 
         :param routine: the routine that class cls calls.
         :type routine: py:class:`psyclone.psyir.symbols.RoutineSymbol`
-        :param arguments: the arguments to this routine. These are \
-            added as child nodes.
-        :type arguments: list of :py:class:`psyclone.psyir.nodes.DataNode`
+        :param arguments: the arguments to this routine, and/or \
+            2-tuples containing an argument name and the \
+            argument. Arguments are added as child nodes.
+        :type arguments: list of \
+            :py:class:`psyclone.psyir.nodes.DataNode` or \
+            (str, :py:class:`psyclone.psyir.nodes.DataNode`)
+
+        :returns: an instance of cls.
+        :rtype: :py:class:`psyclone.psyir.nodes.Call` or a subclass thereof.
 
         :raises GenerationError: if the routine argument is not a \
             RoutineSymbol.
         :raises GenerationError: if the arguments argument is not a \
             list.
-
-        :returns: an instance of cls.
-        :rtype: :py:class:`psyclone.psyir.nodes.Call` or a subclass thereof.
+        :raises GenerationError: if the contents of the arguments \
+            argument are not the expected type.
 
         '''
         if not isinstance(routine, RoutineSymbol):
             raise GenerationError(
-                "Call create routine argument should be a RoutineSymbol but "
-                "found '{0}'.".format(type(routine).__name__))
+                f"Call create routine argument should be a RoutineSymbol but "
+                f"found '{type(routine).__name__}'.")
         if not isinstance(arguments, list):
             raise GenerationError(
-                "Call create arguments argument should be a list but found "
-                "'{0}'.".format(type(arguments).__name__))
+                f"Call create arguments argument should be a list but found "
+                f"'{type(arguments).__name__}'.")
+        args = []
+        names = []
+        for arg in arguments:
+            name = None
+            if isinstance(arg, tuple):
+                if not len(arg) == 2:
+                    raise GenerationError(
+                        f"If a child of the children argument in create "
+                        f"method of Call class is a tuple, it's "
+                        f"length should be 2, but found {len(arg)}.")
+                if not isinstance(arg[0], str):
+                    raise GenerationError(
+                        f"If a child of the children argument in create "
+                        f"method of Call class is a tuple, its first "
+                        f"argument should be a str, but found "
+                        f"{type(arg[0]).__name__}.")
+                name, arg = arg
+            names.append(name)
+            args.append(arg)
 
         call = cls(routine)
-        call.children = arguments
+        call.children = args
+        call._named_args = names
```

Comment:
OK. But does it matter when the access is within the same class as the defined internal variable?

Reply:
*see discussion below about keeping named_argument consistent with children length. Depending how we solve that we don't need a setter because we could do:
```
for idx, name in enumerate(names):
    call.named_args[idx] = name
```

Reply:
The code has been restructured.